Multisig/UI/ClaimToken/ClaimingAmountViewController/ClaimTokensViewController.swift

```diff
@@ -0,0 +1,496 @@
+
+                let ecosystemHighlight = NSAttributedString(string: ecosystemAmount, attributes: [.font: UIFont.systemFont(ofSize: 16, weight: .semibold)])
+                titleTooltipText.replaceCharacters(in: (titleTooltipText.string as NSString).range(of: "$ecosystem"), with: ecosystemHighlight)
+
+                cell.titleTooltipText = titleTooltipText
+
+                // must be set at the end to update values
+                cell.style = .darkGuardian
+            } else if userAllocation != nil, claimData.allocationsData.count == 1 {
+                cell.valueText = formatted(amount: claimData.totalAvailableAmount(of: claimData.allocationsData, at: timestamp))
+                cell.headerTooltipText = nil
+                cell.titleTooltipText = NSAttributedString(string: "Not eligible for Safe Guardian allocation. Contribute to the community to become a Safe Guardian.")
+
+                // must be set at the end to update values
+                cell.style = .darkUser
+            } else {
+                assertionFailure("Data misconfiguration: user or ecosystem allocations not found")
+                cell.valueText = "n/a"
+                cell.headerTooltipText = nil
+                cell.titleTooltipText = nil
+
+                // must be set at the end to update values
+                cell.style = .darkUser
+            }
+            return cell
+
+        case .claimableFuture:
+            let cell = tableView.dequeueCell(AllocationBoxCell.self)
+            cell.headerText = "Claim in the future (vesting)"
+            cell.titleText = "Total"
+            cell.tooltipHostView = view
+
+            guard let claimData = claimData else {
+                // defaults when data not loaded
+                cell.valueText = "..."
+                cell.titleTooltipText = nil
+                cell.headerTooltipText = nil
+
+                // must be set at the end to update values
+                cell.style = .lightUser
+                return cell
+            }
+
+            let userAllocation = claimData.allocationsData.first {
+                $0.allocation.tag.contains("user")
+            }
+            let ecosystemAllocation = claimData.allocationsData.first {
+                $0.allocation.tag.contains("ecosystem")
+            }
+
+            let dateFormatter = DateFormatter()
+            dateFormatter.dateStyle = .short
+            dateFormatter.timeStyle = .short
+
+
+            if let userAllocation = userAllocation, let ecosystemAllocation = ecosystemAllocation, claimData.allocationsData.count == 2 {
+
+                let userAmount = formatted(amount: claimData.unvestedAmount(for: userAllocation, at: timestamp))
+                let ecosystemAmount = formatted(amount: claimData.unvestedAmount(for: ecosystemAllocation, at: timestamp))
+
+                let halfDate = userAllocation.allocation.startDate + 4 * 52 * 7 * 24 * 60 * 60 // 4 years, 52 weeks per year
```

Comment:
question to ask: how to calculate the 4 years, it's hard-coded.

Reply:
question asked, no answer yet. I changed it to use startDate and duration directly.